contracts/utils/contracts/src/Refundable.sol

```diff
@@ -19,6 +19,18 @@
 pragma solidity ^0.5.9;
 
 
-interface IStaking {
-
+contract Refundable {
+
+    // This bool is used by the refund modifier to allow for lazily evaluated refunds.
+    bool internal shouldNotRefund;
+
+    modifier refund {
```

Comment:
Can we rename this to `refundFinalBalance`? It makes the intent of this modifier much more clear IMO.